### PR TITLE
Add CITATION.cff file

### PR DESCRIPTION
**Pull request by matt-graham**
Part of #330 

Adds a CITATION.cff file for this repository with all current contributors included as authors. 
The ordering of authors should match the non-bot accounts on https://github.com/UCL-ARC/python-tooling/graphs/contributors

I have added ORCIDs for everyone I could find them for either from GitHub profile or ARC people directory. Similarly I have tried to enter given and family names as listed on GitHub Profile or ARC people directory but add a review comment if I've made any mistakes or you would prefer to be listed differently.

I am requesting a review from everyone who is listed as an author. If you are happy to be included as an author and don't want any changes made to your information please approve. If you need any changes made to your information or want to be removed as an author please add a review and request changes accordingly.

As I ran out of reviewer slots (maximum of 15 😅) I'm manually tagging @paddyroddy and @samcunliffe

### CODE REVIEW

**Comment by matt-graham** [comment]
@sfmig @adamltyson @alessandrofelder I've listed everyones' affiliation as UCL but let me know if you would prefer I change this to Sainsbury Wellcome Centre

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,84 @@
+cff-version: "1.2.0"
+message: "If you use this template or the associated website, please cite it using these metadata."
```
Comment (matt-graham) [inline CITATION.cff]: At the moment this asks people to use this citation for both template _and_ website. In #330 we discussed possibly making website separately citeable so we may want to change this.

**Review by dstansby — APPROVED** [review]

File: CITATION.cff (hunk 1 above)
Comment (paddyroddy) [reply CITATION.cff]: If we do then we could use the `preferred-citation` syntax https://github.com/astro-informatics/sleplet/blob/e3dec589a7f9803100c9aff6bdefa7ef76a6aa52/CITATION.cff#L10

**Review by paddyroddy — APPROVED** [review]
🫐

**Comment by alessandrofelder** [comment]
> @sfmig @adamltyson @alessandrofelder I've listed everyones' affiliation as UCL but let me know if you would prefer I change this to Sainsbury Wellcome Centre

Given we also don't specify ARC/Mech Eng/... the UCL affiliation seems reasonable to me.

**Review by alessandrofelder — APPROVED** [review]

**Review by nikk-nikaznan — APPROVED** [review]

**Review by t-young31 — APPROVED** [review]

**Review by giordano — APPROVED** [review]

**Review by samcunliffe — APPROVED** [review]

**Review by willGraham01 — APPROVED** [review]

**Review by sfmig — APPROVED** [review]

**Review by adamltyson — APPROVED** [review]

**Review by ruaridhg — APPROVED** [review]

**Review by p-j-smith — APPROVED** [review]

**Comment by p-j-smith** [comment]
> As I ran out of reviewer slots (maximum of 15 😅) I'm manually tagging @paddyroddy and @samcunliffe

Perhaps we should make a `python-tooling` Team?

**Comment by paddyroddy** [comment]
> Perhaps we should make a `python-tooling` Team?

Nice idea, but in this case @matt-graham wants everyone to review it. As soon as one of the team reviewed, the team wouldn't be visible any more.

**Comment by p-j-smith** [comment]
> Nice idea, but in this case @matt-graham wants everyone to review it. As soon as one of the team reviewed, the team wouldn't be visible any more.

Ah yeah, that's a good point. But at least everyone would be notified that they've been requested to review

**Review by katiebuntic — APPROVED** [review]

**Comment by paddyroddy** [comment]
@robertvi @mxochicale @yidilozdemir could you have a look? When @dpshelio gets back, we should be good to go - I'll prod him in person if he hasn't done it

**Review by mxochicale — APPROVED** [review]
Hi Matt and all. 

This looks good to me. 

Perhaps for next iterations: 
* try to add some version release and associated doi. From this [reference](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files): 
```
version: 2.0.4
doi: 10.5281/zenodo.1234
date-released: 2017-12-18
```
* Some family-names and give names have quotes. Not sure how this affects the creation of the citation but I guess it should be some kind of consistency.

Thanks, Miguel

**Comment by matt-graham** [comment]
Thanks for the comments @mxochicale 

>     * try to add some version release and associated doi. From this [reference](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files):
> 
> 
> ```
> version: 2.0.4
> doi: 10.5281/zenodo.1234
> date-released: 2017-12-18
> ```

We have the Zenodo GitHub integration set up already so can add a DOI once we have made a release and a DOI has been minted. In terms of including a version - I would be reticent to have this included unless we had some automated workflow for ensuring this kept up to date with the latest release version as I suspect otherwise it will quickly end up going out of sync.

>     * Some family-names and give names have quotes. Not sure how this affects the creation of the citation but I guess it should be some kind of consistency.

I created the CFF file initially with `cffinit` tool (https://citation-file-format.github.io/cff-initializer-javascript/#/) - it looks like the quotes have been added to names with spaces or non-ASCII characters / characters with accents. We could potentially normalize everything to have quotation marks.

**Comment by paddyroddy** [comment]
> We have the Zenodo GitHub integration set up already so can add a DOI once we have made a release and a DOI has been minted. In terms of including a version - I would be reticent to have this included unless we had some automated workflow for ensuring this kept up to date with the latest release version as I suspect otherwise it will quickly end up going out of sync.

When it comes to the `Zenodo` release, you get a generic DOI (that refers to the whole record) and one for each new version. IMO it's much easier to stick with the generic one, and then you don't need to change anything.

**Comment by matt-graham** [comment]
> When it comes to the `Zenodo` release, you get a generic DOI (that refers to the whole record) and one for each new version. IMO it's much easier to stick with the generic one, and then you don't need to change anything.

Yeah I also usually use the non-version-specific DOI - I was referring above to the suggestion to include a version (and release date) in the `CITATION.cff` itself which is possible but then requires us to keep this up to date with the latest release (and if we're using the non-version specific DOI probably doesn't make sense anyway).

**Review by yidilozdemir — APPROVED** [review]

**Review by robertvi — APPROVED** [review]

**Review by dpshelio — APPROVED** [review]

**Pull request merged**